portfolio/src/main/client/src/util/html.ts

```diff
@@ -0,0 +1,153 @@
+import type {Arrunk} from './types';
+
+type InterpolationValue = Arrunk<string | Node | null | undefined> | EventListener;
+
+/**
+ * Helper function to iterate over all descendants of a node.
+ * @param node The node to iterate over.
+ */
+function* descendants(node: Node): Generator<Node> {
+  for (const child of node.childNodes) {
+    yield child;
+    for (const grandchild of descendants(child)) {
+      yield grandchild;
+    }
+  }
+}
+
+/**
+ * @param arr The array to flatten.
+ * @returns The flattened array.
+ */
+function flatten<T>(arr: Arrunk<T>[]): T[] {
```

Comment:
Use Array.flat instead of writing your own flattener:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat